brainscore/submission/evaluation.py

```diff
@@ -102,48 +121,69 @@ def run_evaluation(config_file, work_dir, jenkins_id, db_secret, models=None,
-    if 'sample-model-submission' in candidates:
-        return 'sample-model-submission'
-    return candidates[0]
-
-
-def clone_repo(config, work_dir):
-    git.Git(work_dir).clone(config['git_url'])
-    return Path('%s/%s' % (work_dir, os.listdir(work_dir)[0]))
-
-
-def install_project(repo, package):
-    try:
-        assert 0 == subprocess.call([sys.executable, "-m", "pip", "install", "-v", repo], env=os.environ)
-        sys.path.insert(1, str(repo))
-        logger.info(f'System paths {sys.path}')
-        return import_module(package)
-    except ImportError:
-        return __import__(package)
+def get_ml_pool(test_models, module, submission):
+    ml_brain_pool = {}
+    if submission.model_type == 'BaseModel':
+        logger.info(f"Start working with base models")
+        layers = {}
+        base_model_pool = {}
+        for model in test_models:
+            function = lambda model_inst=model.name: module.get_model(model_inst)
+            base_model_pool[model.name] = LazyLoad(function)
+            try:
+                layers[model.name] = module.get_layers(model.name)
+            except Exception:
+                logging.warning(f'Could not retrieve layer for model {model} -- skipping model')
+        model_layers = ModelLayers(layers)
+        ml_brain_pool = MLBrainPool(base_model_pool, model_layers)
+    else:
+        logger.info(f"Start working with brain models")
+        for model in test_models:
+            ml_brain_pool[model.name] = module.get_model(model.name)
+    return ml_brain_pool
+
+
+def get_benchmark_instance(benchmark):
+    bench = benchmark_pool[benchmark]
+    benchmark_type, created = BenchmarkType.get_or_create(identifier=benchmark, order=999)
+    if created:
+        try:
+            parent = BenchmarkType.get(identifier=bench.parent)
+            benchmark_type.parent = parent
+            benchmark_type.save()
+        except DoesNotExist:
+            logger.error(
+                f'Couldn\'t connect benchmark {benchmark} to parent {bench.parent} since parent doesn\'t exist')
+        if hasattr(bench, 'bibtex') and bench.bibtex is not None:
+            bibtex_string = bench.bibtex
+            benchmark_type.reference = get_reference(bibtex_string)
+            benchmark_type.save()
+    bench_inst, created = BenchmarkInstance.get_or_create(benchmark=benchmark_type, version=bench.version)
+    if created:
+        # the version has changed and the benchmark instance was not yet in the database
+        ceiling = bench.ceiling
+        bench_inst.ceiling = ceiling.sel(aggregation='center')
+        bench_inst.ceiling_error = ceiling.sel(aggregation='error')
+        bench_inst.save()
+    return bench_inst
+
+
+def get_reference(bibtex_string):
+    parsed = bibtexparser.loads(bibtex_string)
+    entry = list(parsed.entries)[0]
+    ref, create= Reference.get_or_create(bibtex=bibtex_string, author=entry['author'], url=entry['url'], year=entry['year'])
```

Comment:
bit pedantic but: autoformat, there's a missing space before the `=`

Reply:
also does this deal with the case where a reference does not provide the url? (or do we force every benchmark to have that? then we should add a small unit test to check for url existence in each benchmark's bibtex)

Reply:
I am checking that beforehand and only call this if the method is available